### ISSUE TITLE
Better integration of permissions

### ISSUE THREAD

**Issue by raphaelm** [pretix/pretix#39]
- [x] Hide links the users has no permission for
- [x] Add an editor for event permissions
- [x] Add an editor for organizer permissions
- [x] Add a _nice_ editor for teams/permissions
- [x] Restructure permissions such that plugins can define them

**Comment by raphaelm** [comment]
possible duplicate of #123

**Comment by fabian-marquardt** [comment]
Hi @raphaelm, are there currently any plans to implement custom permissions which plugins can define and require? This would be a very useful feature if the functionality provided by a plugin should only be available to a subset of the event organizers. Currently plugin views can only depend on the default set of permissions, which is not sufficient in some cases.

**Comment by raphaelm** [comment]
No, there are currently no such plans for the forseeable future.

We know that a more fine-grained permission system is something lots of people want. However, we haven't yet been able to come up with a convincing design.

Currently, there are 12 permissions in the system. The number has grown only by 1-2 per year so far. If we open the floodgates of making this more fine-grained and allowing plugins to define more permissions, we'd **very likely and quickly** not end up with ~25 in a typical installation, but rather >500 permissions, making it much harder for users to configure them and much easier to get it wrong. Even if we build a great UI to configure them, we're adding new useful features to pretix every month. If we'd do permissions on a fine level, we'd also need to add new permissions every month, causing the new features to not show up for users until they manually re-configure the permissions (since we can hardly guess what they want).

**Comment by fabian-marquardt** [comment]
Thank you for the swift and detailed response. I totally agree that such a redesigned permission system needs to be based on a well thought out concept and this will certainly not be done easily!
